### PR TITLE
MBConv C Implementation

### PR DESCRIPTION
**Pull request by ShikharJ**
Yet another Work In Progress PR. This isn't ready yet, To-dos are as follows:

- [x] Complete C Implementation (with better naming conventions)
- [x] Implement Tests For Each Function Introduced
- [x] Provide Extensive Documentation

### CODE REVIEW

File: c_reference/include/quantized_mbconv.h [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ * @param[in]        shlU1          scale to multiply with the first TreeSum output
+ * @param[in]        shlB1          scale to multiply with the first BatchNorm addition factor
+ * @param[in]        shlX1          scale to multiply with the first Convolution output
+ * @param[in]        shlU2          scale to multiply with the second TreeSum output
+ * @param[in]        shlB2          scale to multiply with the second BatchNorm addition factor
+ * @param[in]        shlX2          scale to multiply with the second Convolution output
+ * @param[in]        shlU3          scale to multiply with the third TreeSum output
+ * @param[in]        shlB3          scale to multiply with the third BatchNorm addition factor
+ * @param[in]        shlW3          scale to multiply with the third Convolution output
+ */
```
Comment (harsha-simhadri) [inline c_reference/include/quantized_mbconv.h:64]: Add description of the math behind this operator

File: c_reference/include/quantized_utils.h [hunk 2]
```diff
@@ -20,7 +20,21 @@ inline INT_T saturate(INTM_T inp) {
...
+inline INTM_T q_relu(INTM_T inp, INTM_T limit) {
+    if (inp > limit){
+        return limit;
+    } else if (inp < 0) {
+        return 0;
+    } else {
+        return inp;
+    }
+}
+
```
Comment (harsha-simhadri) [inline c_reference/include/quantized_utils.h:34]: check with Arun's PR whether you are redefining relu

File: c_reference/src/quantized_utils.c [hunk 3]
```diff
@@ -3,6 +3,45 @@
 
 #include "quantized_utils.h"
 
+void v_q_treesum(INTM_T* const vec, ITER_T len, SCALE_T H1, SCALE_T H2) {
```
Comment (harsha-simhadri) [inline c_reference/src/quantized_utils.c:6]: check conflict with Arun's PR

File: c_reference/tests/utils/test_quantized_utils.c [hunk 4]
```diff
@@ -18,6 +18,15 @@ int check_output(const INT_T* const pred, const INT_T* const expected,
   return 0;
 }
 
+// Test v_q_treesum() function.
+int test_v_q_treesum() {
+  INTM_T qvec_A[8] = {-425, -169, -3534, 524, -2739, 87, 52, 292};
+  const INTM_T expected[1] = {-738};
```
Comment (harsha-simhadri) [inline c_reference/tests/utils/test_quantized_utils.c:24]: can we try to input a full activation map corresponding to an image.

File: c_reference/include/quantized_utils.h (hunk 2 above)
Comment (ShikharJ) [reply c_reference/include/quantized_utils.h:34]: Arun's PR doesn't take a limit variable for `ReLU`, plus it takes an entire `INT_T` matrix as an input. I think the better thing to do here is to merge this as is, and make further modifications when we're rebasing Arun's commits to master.

File: c_reference/src/quantized_utils.c (hunk 3 above)
Comment (ShikharJ) [reply c_reference/src/quantized_utils.c:6]: No conflict here.

File: c_reference/tests/utils/test_quantized_utils.c (hunk 4 above)
Comment (ShikharJ) [reply c_reference/tests/utils/test_quantized_utils.c:24]: Adding the convolution output of the third filter in MBConv (obtained from SeeDot's code), as a test.

**Pull request merged**